parsing/parser.mly

```diff
@@ -2388,8 +2381,10 @@ val_longident:
 ;
 constr_longident:
     mod_longident       %prec below_DOT         { $1 }
+  | mod_longident DOT LPAREN COLONCOLON RPAREN  { Ldot($1,"::") }
```

Comment:
What about `mod_longident DOT LPAREN RPAREN` and `mod_longident DOT LBRACKET RBRACKET`?